bin/oref0-bash-common-functions.sh

```diff
@@ -0,0 +1,431 @@
+# information and exit with status 0 (success). Otherwise, save the usage
+# information in environment variable HELP_TEXT so it can be used by print_usage
+# later.
+#
+# Correct invocation would look like:
+#    usage "$@" <<EOT
+#    Usage: $(basename $0) [--some-argument] [--some-other-argument]
+#    Description of what this tool does. Information about what the arguments do.
+#    EOT
+usage () {
+    case "$1" in
+        help|-h|--help)
+            cat -
+            exit 0
+            ;;
+    esac
+    export HELP_TEXT=$(cat -)
+}
+
+# Print the program's help text, as previously set by usage(). This would
+# typically be used after detecting invalid arguments, and followed by "exit 1".
+print_usage () {
+    echo "$HELP_TEXT"
+}
+
+# Check that the current working directory contains openaps.ini, ie, is an
+# OpenAPS session directory. This is presumably the myopenaps directory (though
+# in principle it could also be ~/myopenaps-cgm-loop or something not part of
+# the standard install). If it isn't, print a message saying it should be run
+# from ~/myopenaps to stderr and exit with status 1 (failure).
+assert_cwd_contains_ini () {
+    if [[ ! -e "openaps.ini" ]]; then
+        echo "$self: This script should be run from the myopenaps directory, but was run from $PWD which does not contain openaps.ini." 1>&2
+        exit 1
+    fi
+}
+
+
+# Returns success (0) if running on an Intel Edison, fail (1) otherwise. Uses
+# the existence of an "edison" account in /etc/passwd to determine that.
+is_edison () {
+    if getent passwd edison >/dev/null; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Returns success (0) if running on a Raspberry Pi, fail (1) otherwise. Uses
+# the existence of a "pi" account in /etc/passwd to determine that.
+is_pi () {
+    if getent passwd pi > /dev/null; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Usage: file_is_recent <path> <minutes>
+# Return success if the modification time of the file at the given path has a
+# modification time newer than the given number of minutes (default 5). Returns
+# failure if the file doesn't exist, or is old.
+file_is_recent () {
+    BASE_NAME="$(basename $1)"
+    find "$(dirname $1)" -mmin -"${2-5}" -name "$BASE_NAME" |grep -q "$BASE_NAME" >/dev/null 2>&1
+    return $?
+}
+
+# Usage: file_is_recent_and_min_size <path> <minutes> <bytes>
+# Return success if the modification time of the file at the given path has a
+# modification time newer than the given number of minutes (default 5) *and* is
+# at least the given number of bytes (default 5). The minimum size is chosen
+# to exclude files like "{}\n" but include just about everything else. Returns
+# failure if the file doesn't exist, is too small, or is old.
+file_is_recent_and_min_size () {
+    BASE_NAME="$(basename $1)"
+    find "$(dirname $1)" -mmin -"${2-5}" -name "$BASE_NAME" -size +${3-5}c |grep -q "$BASE_NAME" >/dev/null 2>&1
```

Comment:
This is existing, but i'm not sure why we need to search for the BASE_NAME here instead of just look for the file by full name? Also that extra grep seems unnecessary.

Reply:
`find` is being used in a slightly weird way here; what it does is it recursively enumerates files in `$(dirname $1)` and outputs to stdout the name of every file which matches the `-mmin` and `-name` filters. `find` always returns true, so the `grep` checks whether the file we're looking for was in the output.